Application/EdFi.Ods.Api/Controllers/TokenController.cs

```diff
@@ -36,8 +38,60 @@ public TokenController(ITokenRequestProvider provider)
         [HttpPost]
         [AllowAnonymous]
         [Consumes("application/json")]
-        public async Task<IActionResult> Post([FromBody] TokenRequest tokenRequest)
+        public async Task<IActionResult> PostFromJsonAsync([FromBody] TokenRequest tokenRequest)
         {
+            // Look for the authorization header, since we MUST support this method of authorization
+            // https://tools.ietf.org/html/rfc6749#section-2.3.1
+            // Decode and parse the client id/secret from the header
+            // Authorization is in a form of Bearer <encoded client and secret>
+
+            if (!Request.Headers.ContainsKey("Authorization"))
+            {
+                _logger.Debug($"Header is missing authorization credentials");
+                return Unauthorized();
+            }
+
+            string[] encodedClientAndSecret = Request.Headers["Authorization"]
+                .ToString()
+                .Split(' ');
+
+            if (encodedClientAndSecret.Length != 2)
+            {
+                _logger.Debug("Header is not in the form of Basic <encoded credentials>");
+                return Unauthorized();
+            }
+
+            if (!encodedClientAndSecret[0]
+                .EqualsIgnoreCase("Basic"))
+            {
+                _logger.Debug("Authorization scheme is not Basic");
+                return Unauthorized(new TokenError(TokenErrorType.InvalidClient));
+            }
+
+            string[] clientIdAndSecret;
+
+            try
+            {
+                clientIdAndSecret = Encoding.UTF8.GetString(Convert.FromBase64String(encodedClientAndSecret[1]))
+                    .Split(':');
```

Comment:
lets make a private method for this please that both methods can use

Reply:
Fixed